### ISSUE TITLE
proxied via https support

### ISSUE THREAD

**Issue by tkurki** [SignalK/freeboard-sk#72]
Apparently the protocol is hard coded to ws - should take into account also the protocol where it was loaded from.

**Comment by panaaj** [comment]
Not sure to what scenario you are referring to which this happens.
All connections (api or stream) are based on the received endpoints from the hello message from the server.
I cannot replicate a situation where `https` or `wss` is not used if the endpoints define it.

If you could let me know how you get the issue to appear I will try to replicate it.

**Comment by tkurki** [comment]
Oh right. My setup is with balena.io, that provides a publicly accessible proxied https connection to the http server running on the RPi. The server does not know that it is proxied via https and nor should it, because that would break local use.

What if freeboard tried to simply fall back to the host it was loaded from with no discovery if discovery fails?

**Comment by panaaj** [comment]
So something like.... 
Send request to `protocol://host:port/signalk`
`If` hello response received `then` use it
`else if` (no response || response does not contain endpoints: {  v1: { signalk-http: {}, signalk-ws: {} } 
`then` fall back to 
```
{ 
   endpoints: { 
      v1: {
         signalk-http= "protocol://host:port/signalk/v1/api/",
         signalk-ws= "protocol://host:port/signalk/v1/stream",
      }
   }
}
```

**Comment by panaaj** [comment]
Addressed in v2.0.0

**Comment by panaaj** [comment]
V2.0.0 released.
